### PR TITLE
[#OSF-8174]Make file view counts through OSF file storage available through the API

### PR DESCRIPTION
**Pull request by chennan47**
## Purpose
Put all previous file view counts work together to a working version which is not affected by MFR caching problem.

## Ticket
https://openscience.atlassian.net/browse/OSF-8174

### CODE REVIEW

File: api_tests/files/views/test_file_detail.py [hunk 1]
```diff
@@ -463,6 +463,32 @@ def test_folder_files_relationships_contains_guid_not_id(self, app, user, node):
...
+        file1.versions.last().reload()
+
+        assert res.status_code == 200
+        assert res.json.keys() == ['data']
+        attributes = res.json['data']['attributes']
+        assert attributes['path'] == file1.path
+        assert attributes['kind'] == file1.kind
+        assert attributes['name'] == file1.name
+        assert attributes['materialized_path'] == file1.materialized_path
+        assert attributes['last_touched'] == None
```
Comment (alexschiller) [inline api_tests/files/views/test_file_detail.py]: Use `is` when comparing against `None`

File: api_tests/files/views/test_file_detail.py [hunk 2]
```diff
@@ -463,6 +463,32 @@ def test_folder_files_relationships_contains_guid_not_id(self, app, user, node):
         assert node._id in split_href
         assert node.id not in split_href
 
+    def test_files_view_count(self, app, user, node):
+        file1 = api_utils.create_test_file(node, user, create_guid=False)
```
Comment (alexschiller) [inline api_tests/files/views/test_file_detail.py]: prefer `file_one`

File: api_tests/files/views/test_file_detail.py [hunk 3]
```diff
@@ -463,6 +463,32 @@ def test_folder_files_relationships_contains_guid_not_id(self, app, user, node):
...
+        file1.save()
+        file1_url = '/{}files/{}/'.format(API_BASE, file1._id)
```
Comment (alexschiller) [inline api_tests/files/views/test_file_detail.py]: prefer `url_file_one` (ordered as "thing it is_description/relationship")

File: addons/base/views.py [hunk 4]
```diff
@@ -664,6 +666,9 @@ def addon_view_or_download_file(auth, path, provider, **kwargs):
             'Location': file_node.generate_waterbutler_url(**dict(extras, direct=None, version=version.identifier, _internal=extras.get('mode') == 'render'))
         }))
 
+    if action == 'view' and provider == 'osfstorage':
+        update_analytics(node_addon.owner, file_node._id, int(version.identifier) - 1, download=False)
```
Comment (sloria) [inline addons/base/views.py]: I think we'll need to remove this when https://openscience.atlassian.net/browse/PLAT-218 after implemented.

File: osf/management/commands/migrate_file_view_counts_from_keen.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+
+def set_file_view_counts():
+    # get all osfstorage files which is_deleted == False, the file size in production database > 1730325
+    files = OsfStorageFile.objects.all()
+
+    # the limit of the datasets size return from keen is 400kb.
+    # A json file return 3 file counts from keen is about 223 bytes
+    # So a json file return from keen should be able to reach 4800 files, to be safe use 4500
+    keen_file_limit = 4500
```
Comment (sloria) [inline osf/management/commands/migrate_file_view_counts_from_keen.py]: Note: We should use Django's `Paginator` or the batch iteration technique that we've used in migrations here: https://github.com/CenterForOpenScience/osf.io/blob/b7b8e9b78348f1fe4b22fca8e6110072dc497aa7/osf/migrations/0103_set_osf_storage_node_settings_region.py#L33-L37

**Comment by pattisdr** [comment]
Closing - going a different route to surface view counts.

**Pull request closed without merging**